sw/device/tests/sysrst_ctrl_ec_rst_l_test.c

```diff
@@ -0,0 +1,138 @@
+// Copyright lowRISC contributors.
+// Licensed under the Apache License, Version 2.0, see LICENSE for details.
+// SPDX-License-Identifier: Apache-2.0
+
+#include "sw/device/lib/base/mmio.h"
+#include "sw/device/lib/dif/dif_pinmux.h"
+#include "sw/device/lib/dif/dif_pwrmgr.h"
+#include "sw/device/lib/dif/dif_rstmgr.h"
+#include "sw/device/lib/dif/dif_sysrst_ctrl.h"
+#include "sw/device/lib/runtime/ibex.h"
+#include "sw/device/lib/runtime/log.h"
+#include "sw/device/lib/testing/rstmgr_testutils.h"
+#include "sw/device/lib/testing/sysrst_ctrl_testutils.h"
+#include "sw/device/lib/testing/test_framework/check.h"
+#include "sw/device/lib/testing/test_framework/ottf_main.h"
+
+#include "hw/top_earlgrey/sw/autogen/top_earlgrey.h"
+
+/* We need control flow for the ujson messages exchanged
+ * with the host in OTTF_WAIT_FOR on real devices. */
+OTTF_DEFINE_TEST_CONFIG(.enable_uart_flow_control = true);
+
+static dif_pinmux_t pinmux;
+static dif_sysrst_ctrl_t sysrst_ctrl;
+static dif_pwrmgr_t pwrmgr;
+static dif_rstmgr_t rstmgr;
+static dif_rstmgr_reset_info_bitfield_t rstmgr_reset_info;
+
+enum {
+  kAllZero = 0x0,
+  kAllOne = 0x3,
+  kNumMioInputs = 0x2,
+};
+
+static const dif_pinmux_index_t kPeripheralInputs[] = {
+    kTopEarlgreyPinmuxPeripheralInSysrstCtrlAonKey0In,
+    kTopEarlgreyPinmuxPeripheralInSysrstCtrlAonKey1In,
+};
+
+static const dif_pinmux_index_t kInputPads[] = {
+    kTopEarlgreyPinmuxInselIor6,
+    kTopEarlgreyPinmuxInselIor7,
+};
+
+// Threshold/Duration values are not specific to a real-world debounce
+// scenario so are kept short to avoid excessive simulation time.
+// Assuming a 5us aon clock period.
+enum {
+  kDetectionTimeThreshold = 1024,  // ~5ms
+  kEcResetStretchDurationMicros = 200 * 1000,
+  kDebounceTimeThreshold = 128,  // ~0.6ms
+};
+
+// Sets up the pinmux to assign input and output pads
+// to the sysrst_ctrl peripheral as required.
+static void pinmux_setup(void) {
+  for (int i = 0; i < kNumMioInputs; ++i) {
+    CHECK_DIF_OK(
+        dif_pinmux_input_select(&pinmux, kPeripheralInputs[i], kInputPads[i]));
+  }
+}
+
+static void configure_combo_reset(void) {
+  // FIXME (pamaury): if I set the ec reset pulse width to 0 or 1,
+  // then on reset the EC reset line pulled down, then up for a short time
+  // and then down again. This seems contrary to the documentation and also
+  // contrary to what happens for values >1 when it is only pulled down.
```

Comment:
(Similar FIXME question, but...): I think it might be better to add a comma: "... values >1, when it ...".

I initially wondered whether you meant that something different happened if both of these things were true:
- Value > 1
- Something is only pulled down